Python/bytecodes.c

```diff
@@ -537,9 +582,12 @@ dummy_func(
         // specializations, but there is no output.
         // At the end we just skip over the STORE_FAST.
         tier1 op(_BINARY_OP_INPLACE_ADD_UNICODE, (left, right --)) {
+            PyObject *left_o = PyStackRef_AsPyObjectBorrow(left);
+            PyObject *right_o = PyStackRef_AsPyObjectBorrow(right);
+
             assert(next_instr->op.code == STORE_FAST);
-            PyObject **target_local = &GETLOCAL(next_instr->op.arg);
-            DEOPT_IF(*target_local != left);
+            _PyStackRef *target_local = &GETLOCAL(next_instr->op.arg);
+            DEOPT_IF(PyStackRef_AsPyObjectBorrow(*target_local) != left_o);
```

Comment:
```suggestion
            DEOPT_IF(!PyStackRef_Is(*target_local, left);
```

Where `PyStackRef_Is(a, b)` is defined as `a.bits == b.bits`.